### PR TITLE
Tidying of identifiers/, lattices/ etc.

### PR DESCRIPTION
**Pull request by mshinwell**
A variety of minor changes to significantly clean things up.  Some highlights:

The `Container_types` hash tables aren't used in the core Flambda 2 code so I've removed them.  The only use was in `compare.ml` which for the moment can just use maps, the speed doesn't seem critical here.

Various places (`Compilation_unit`, `Function_slot` and `Value_slot`) that used to need `Ident`s or `Variable`s to be created only for their names (!) have been fixed.

The implementations of `Function_slot` and `Value_slot` are now shared.

The unwieldy `Reg_width_things` name has been renamed to `Int_ids`.  Various unnecessary qualifications with that module name have also been removed.

The debug option for permuting all names has been removed, it saves a fair amount of code, and we don't have problems in this area any more.  The iteration functions nearby were moved into the Fexpr code, since they're not used in the core Flambda 2 system.

### CODE REVIEW

File: middle_end/flambda2/identifiers/int_ids.mli [hunk 1]
```diff
@@ -15,7 +15,8 @@
 (**************************************************************************)
 
 (** The underlying implementation for [Variable], [Symbol], [Name],
-    [Reg_width_const] and [Simple]. *)
+    [Reg_width_const] and [Simple]. These values are all represented as
+    integers. *)
```
Comment (lthls) [inline middle_end/flambda2/identifiers/int_ids.mli:19]: Note: Code IDs too

File: middle_end/flambda2/lattices/or_bottom.mli [hunk 2]
```diff
@@ -36,4 +36,8 @@ module Let_syntax : sig
   val ( let<* ) : 'a t -> ('a -> 'b t) -> 'b t
 
   val ( let<+ ) : 'a t -> ('a -> 'b) -> 'b t
+
+  (** [let<+$] returns the input ['a t] if the mapping function returns a value
+      that is physically-equal to its input. *)
+  val ( let<+$ ) : 'a t -> ('a -> 'a) -> 'a t
```
Comment (lthls) [inline middle_end/flambda2/lattices/or_bottom.mli:42]: This operator is not even used. Do we really need it ?

**Review by poechsel — APPROVED** [review]
I approve my part of the review

File: middle_end/flambda2/lattices/or_bottom.mli (hunk 2 above)
Comment (mshinwell) [reply middle_end/flambda2/lattices/or_bottom.mli:42]: I thought it might be better to have it for consistency, but happy to remove it.

**Comment by mshinwell** [comment]
I'm reverting the ocamlformat attribute changes on print functions that have any non-trivial formatting string.

File: middle_end/flambda2/identifiers/int_ids.mli (hunk 1 above)
Comment (mshinwell) [reply middle_end/flambda2/identifiers/int_ids.mli:19]: Fixed

**Review by lthls — APPROVED** [review]

**Pull request merged**